### PR TITLE
Add command joke

### LINKED ISSUE #15: Add command `joke`
Print a random joke about PHP

---

Or print a joke of the day on start

### PR DESCRIPTION
**Pull request by k2gl**
## What was changed

Added a new console command `joke` that emits a random joke via `trap()` function

## Checklist

- Closes #15
- Tested
  - [x] Tested manually
  - [ ] Unit tests added
- [ ] Documentation

### CODE REVIEW

File: src/Cli/StartupJokeSpeechBubble.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    public const JOKES = [
+        'Why do programmers always mix up Halloween and Christmas? Because Oct 31 == Dec 25.',
+        'Two hard things in computer science: cache invalidation, naming things and stack overflow.',
+        'Depressive programming style through dump and die.',
+        'PHP was dead 84 years ago right?',
+    ];
+
+    public static function randomStartupJokeSpeechBubble(): string
+    {
+        $joke = self::JOKES[mt_rand(0, \count(self::JOKES) - 1)];
```
Comment (roxblnfk) [inline src/Cli/StartupJokeSpeechBubble.php:18]:
```suggestion
        $joke = self::JOKES[\mt_rand(0, \count(self::JOKES) - 1)];
```

It's preffered to use FQN for functions from global scope

File: src/Cli/StartupJokeSpeechBubble.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Buggregator\Trap\Cli;
+
+class StartupJokeSpeechBubble
+{
+    public const JOKES = [
+        'Why do programmers always mix up Halloween and Christmas? Because Oct 31 == Dec 25.',
+        'Two hard things in computer science: cache invalidation, naming things and stack overflow.',
+        'Depressive programming style through dump and die.',
+        'PHP was dead 84 years ago right?',
```
Comment (roxblnfk) [inline src/Cli/StartupJokeSpeechBubble.php:13]: @butschster are there any Buggregator related phrases and jokes in your baggage?

File: src/Cli/StartupJokeSpeechBubble.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        $joke = self::JOKES[mt_rand(0, \count(self::JOKES) - 1)];
+        $jokeLen = max(\strlen($joke), 13);
+        $speechBubbleLeftSpace = str_repeat(' ', 14);
+
+        $lines = [];
+        $lines[] = $speechBubbleLeftSpace . '⠘⡀⠀⠀' . str_repeat(' ', $jokeLen) . '⠀⠀ ⡜';
+        $lines[] = $speechBubbleLeftSpace . ' ⠘⡀⠀' . str_pad($joke, $jokeLen, ' ', STR_PAD_BOTH) . '⠀ ⡜';
+        $lines[] = $speechBubbleLeftSpace . '  ⠑⡀' . str_repeat(' ', $jokeLen) . '⡔⠁';
+        $lines[] = $speechBubbleLeftSpace . '   ⢸    ' . str_repeat('⣀', $jokeLen - 13) . '⣀⣀⣀⣀⡀⠤⠄⠒⠈';
+        $lines[] = $speechBubbleLeftSpace . '   ⠘⣀⠄⠊⠁';
```
Comment (roxblnfk) [inline src/Cli/StartupJokeSpeechBubble.php:27]: It's a fun idea. Need to compare with another design solutions.

Also need to add a console command to run a joke `vendor/bin/trap joke`.
The jok may be print via `trap($joke)` call. The if a Buggregator instance is run, the joke will be shown there.

File: src/Command/Joke.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+namespace Buggregator\Trap\Command;
+
+use Buggregator\Trap\Info;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Run application
```
Comment (roxblnfk) [inline src/Command/Joke.php:14]: Incorrect description

File: src/Command/Joke.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    name: 'joke',
+    description: 'Print joke',
+)]
+final class Joke extends Command
+{
+    protected function execute(
+        InputInterface $input,
+        OutputInterface $output,
+    ): int {
+        trap(Info::JOKES[\array_rand(Info::JOKES)]);
```
Comment (roxblnfk) [inline src/Command/Joke.php:28]:
```suggestion
        \trap(Info::JOKES[\array_rand(Info::JOKES)]);
```

File: src/Command/Run.php [hunk 6]
```diff
@@ -55,7 +55,8 @@ protected function execute(
     ): int {
         try {
             // Print intro
-            $output->writeln(\sprintf('<fg=yellow;options=bold>%s</> <info>v%s</>', Info::NAME, Info::VERSION));
+            $output->writeln(\sprintf('<fg=yellow;options=bold>%s</> <info>v%s</>:', Info::NAME, Info::VERSION));
+            trap(Info::JOKES[\array_rand(Info::JOKES)]);
```
Comment (roxblnfk) [inline src/Command/Run.php:59]:
```suggestion
            $output->writeln(\sprintf('<fg=yellow;options=bold>%s</> <info>v%s</>', Info::NAME, Info::VERSION));
```

I think we don't really need to print jokes every run

File: src/Info.php [hunk 7]
```diff
@@ -27,4 +27,11 @@ class Info
         CONSOLE;
 
     public const TRAP_ROOT = __DIR__ . '/..';
+    public const JOKES = [
+        'Why do programmers always mix up Halloween and Christmas? Because Oct 31 == Dec 25.',
+        'Two hard things in computer science: cache invalidation, naming things and stack overflow.',
+        'Depressive programming style through dump and die.',
+        'PHP was dead 84 years ago right?',
+        'Submit a pull request to help us improve the Buggregator Trap codebase',
+    ];
```
Comment (roxblnfk) [inline src/Info.php:36]: Please move it into `resources/registry/jokes.txt` file.
Format: base64 encoded joke text per line.

Why encoded?
- not to spoiler all the messages
- support multiline texts

It's OK not to make a service, and just implement a reader inside the command.

File: src/Command/Joke.php (hunk 4 above)
Comment (k2gl) [reply src/Command/Joke.php]: fixed

File: src/Command/Joke.php (hunk 5 above)
Comment (k2gl) [reply src/Command/Joke.php]: fixed

File: src/Command/Run.php (hunk 6 above)
Comment (k2gl) [reply src/Command/Run.php]: deleted

File: src/Info.php (hunk 7 above)
Comment (k2gl) [reply src/Info.php:36]: done

**Comment by roxblnfk** [comment]
LGTM
Now we need a list of good jokes and funny phrases.
@butschster, can you help us with this?

**Pull request merged**